packages/safe-apps-provider/src/provider.ts

```diff
@@ -7,44 +9,44 @@ const NETWORK_CHAIN_ID: Record<string, number> = {
   XDAI: 100,
 };
 
-// taken from ethers.js, compatible interface with web3 provider
-type AsyncSendable = {
-  isMetaMask?: boolean;
-  host?: string;
-  path?: string;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  sendAsync?: (request: any, callback: (error: any, response: any) => void) => void;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  send?: (request: any, callback: (error: any, response: any) => void) => void;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  request?: (request: { method: string; params?: Array<any> }) => Promise<any>;
-};
-
-export class SafeAppProvider implements AsyncSendable {
+// The API is based on Ethereum JavaScript API Provider Standard. Link: https://eips.ethereum.org/EIPS/eip-1193
+export class SafeAppProvider implements EIP1193Provider {
   private readonly safe: SafeInfo;
   private readonly sdk: SafeAppsSDK;
   private submittedTxs = new Map<string, Web3TransactionObject>();
+  private events = new EventEmitter();
 
   constructor(safe: SafeInfo, sdk: SafeAppsSDK) {
     this.safe = safe;
     this.sdk = sdk;
   }
 
-  public get chainId(): number {
-    return NETWORK_CHAIN_ID[this.safe.network];
+  async connect(): Promise<void> {
+    return;
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  sendAsync(request: any, callback: (error: any, response: any) => void): void {
-    this.send(request, callback);
+  async disconnect(): Promise<void> {
+    return;
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  send(request: any, callback: (error: any, response?: any) => void): void {
-    if (!request) callback('Undefined request');
-    this.request(request)
-      .then((result) => callback(null, { jsonrpc: '2.0', id: request.id, result }))
-      .catch((error) => callback(error, null));
+  public on(event: string, listener: any): void {
```

Comment:
Isn't the listener always a function?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/events.d.ts#L63
We could probably import the entire type definition for the EventEmitter.

Reply:
good idea :)